spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -0,0 +1,276 @@
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to edit registration page' do
+          expect(response).to redirect_to edit_conference_conference_registrations_path(conference_with_open_registration.short_title)
+        end
+      end
+
+      context 'ichain is enabled and user is not signed in' do
+        before do
+          CONFIG['authentication']['ichain']['enabled'] = true
+          sign_out user
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        after { CONFIG['authentication']['ichain']['enabled'] = false }
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+      end
+
+      context 'registration limit has reached' do
+        before do
+          conference_with_open_registration.update_attributes(registration_limit: 1)
+          create(:registration, conference: conference_with_open_registration)
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'redirects to root path' do
+          expect(response).to redirect_to root_path
+        end
+
+        it 'shows error in flash message' do
+          expect(flash[:alert]).to match "Sorry, registration limit exceeded for #{conference_with_open_registration.title}"
+        end
+      end
+
+      context 'successful request' do
+        before do
+          get :new, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'assigns registration and user variables' do
+          expect(assigns(:registration)).to be_instance_of(Registration)
+          expect(assigns(:user)).to be_instance_of(User)
+        end
+
+        it 'renders the new template' do
+          expect(response).to render_template('new')
+        end
+      end
+    end
+
+    describe 'GET #show' do
+      before do
+        @registration = create(:registration, conference: conference_with_open_registration, user: user)
+      end
+
+      context 'user has purchased a ticket' do
+        before do
+          @ticket = create(:ticket, conference: conference_with_open_registration)
+          @purchased_ticket = create(:ticket_purchase, conference: conference_with_open_registration,
+                                                       user: user,
+                                                       ticket: @ticket)
+          get :show, conference_id: conference_with_open_registration.short_title
+        end
+
+        it 'renders the show template' do
+          expect(response).to render_template('show')
+        end
+
+        it 'assigns workshops variable' do
+          expect(assigns(:workshops)).to eq @registration.workshops
+        end
+
+        it 'assigns price of purchased tickets to total_price and purchased tickets to tickets' do
+          expect(assigns(:total_price)).to eq @ticket.price * @purchased_ticket.quantity
```

Comment:
Make it explicit. Instead of the multiplication, simply put the right number here.